micropython/usb/examples/device/midi_example.py

```diff
@@ -0,0 +1,50 @@
+# MicroPython USB MIDI example
+#
+# This example demonstrates creating a custom MIDI device.
+#
+# This example uses the usb-device-midi package for the MIDIInterface class.
+# This can be installed with:
+#
+# mpremote mip install usb-device-midi
+#
+# MIT license; Copyright (c) 2023-2024 Angus Gratton
+import usb.device
+from usb.device.midi import MIDIInterface
```

Comment:
I'm unsure whether it's better to have each sub-package install its own module like this, or to have the top-level usb/device/__init__.py expose all of the public interfaces in that package (in which case it would be `usb.device.MIDIInterface`).

* Putting everything in the top `usb.device` namespace may be a little better for discoverability, as can tab-complete all the "public" class names.
* It seems like no matter what the filesystem will expose the `usb.device.midi` module, so the current version clutters the namespace less overall.
* Relying on the filesystem to discover the separate modules makes the binary code size smaller as `__init__.py` is nice and small. (On the other hand, I'm not sure if relying on the filesystem to expose these modules is a good practice.)
* Some modules contain more than one "public" class, for example `usb.device.keyboard` contains `KeyboardInterface`, `KeyCode`, and `LEDCode`. So namespacing these together in their own keyboard module seems sensible.

Reply:
Discussed this with @dpgeorge just now, and looks like we'll keep this approach for USB even though it's a little different to lora & aioble.